File: 1431-kids-with-the-greatest-number-of-candies/1431-kids-with-the-greatest-number-of-candies.py
class Solution:
    def kidsWithCandies(self, candies: List[int], extraCandies: int) -> List[bool]:
        max_possesion = max(candies)
        
        result = []
        for candy in candies:
            if candy + extraCandies >= max_possesion:
                result.append(True)
            else: 
                result.append(False)
        
        return result

File: 1431-kids-with-the-greatest-number-of-candies/1431-kids-with-the-greatest-number-of-candies.go
func kidsWithCandies(candies []int, extraCandies int) []bool {
    l := len(candies)

    greatest := slices.Max(candies)
    rslt := make([]bool, l)
    
    for i:=0; i<l; i++ {
        total := candies[i] + extraCandies
        if total >= greatest{
            rslt[i] = true
        }else{
            rslt[i] = false
        }
    }

    return rslt
}